src/backend/commands/typecmds.c

```diff
@@ -455,8 +455,34 @@ DefineType(ParseState *pstate, List *names, List *parameters)
 					(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
 					 errmsg("storage \"%s\" not recognized", a)));
 	}
-	if (collatableEl)
-		collation = defGetBoolean(collatableEl) ? DEFAULT_COLLATION_OID : InvalidOid;
+	if (collatableEl && defGetBoolean(collatableEl))
+	{
+		/*
+		 * For Babelfish, we are creating new type duting create extension time under sys schema,
+		 * We want these datatype being created under sys schema to have the correct collation.
+		 */
```

Comment:
To minimize engine changes, can we use a hook here? e.g.,
```
	if (collatableEl)
		collation = defGetBoolean(collatableEl) ? DEFAULT_COLLATION_OID : InvalidOid;

    if (defineTypeCollHook)
        collation = defineTypeCollHook(collation, typeNamespace);
```

Reply:
Good idea.. Moved it inside hook